### PR TITLE
updates for jenkins build scripts

### PR DESCRIPTION
**Pull request by akva2**
- add a generic function to build upstream modules
- add a generic function to build downstream modules
- add downstream trigger support in this module

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this with downstreams please

File: jenkins/build.sh [hunk 1]
```diff
@@ -6,12 +6,66 @@ source `dirname $0`/build-opm-module.sh
...
+             opm-simulators
+             opm-upscaling)
+declare -A downstreamRev
+downstreamRev[opm-parser]=master
+downstreamRev[opm-material]=master
+downstreamRev[opm-core]=master
+downstreamRev[opm-grid]=master
+downstreamRev[opm-output]=master
+downstreamRev[opm-simulators]=master
+downstreamRev[opm-upscaling]=master
```
Comment (akva2) [inline jenkins/build.sh]: note to self: ewoms missing

**Comment by akva2** [comment]
@bska @atgeirr @joakim-hove ready for review.

**Comment by joakim-hove** [comment]
My personal "shell script too long" limit is about four lines; so this essentially blows the scale. Unless someone objects before lunch tomorrow I will merge this on @akva2 historical track record.

**Comment by atgeirr** [comment]
> I will merge this on @akva2 historical track record.

Seconded. Unless @akva2 wanted to add ewoms first? (Re: comment in code.)

**Comment by akva2** [comment]
i fixed that the morning after. it's in there now.

**Pull request merged**